xhr/open-parameters-toString.htm

```diff
@@ -0,0 +1,54 @@
+<!doctype html>
+<title>XMLHttpRequest: open() attempts to toString its string parameters</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<div id="log"></div>
+<script>
+test(() => {
+  let log = [];
+  let expected = [
+    'method',
+    'url',
+    // NOTE: 'async' intentionally missing
```

Comment:
It'd be more helpful to comment on why it's missing. It seems that per https://tc39.github.io/ecma262/#sec-toboolean which is used for boolean conversion it'd end up being true because you passed an object. The note below therefore seems wrong too...

Reply:
I think it's missing not because it is `ToBoolean`'d and `ToBoolean` doesn't invoke `valueOf`, but because of [the `open` algorithm](https://xhr.spec.whatwg.org/#the-open()-method), particularly step 5

> If the async argument is omitted, set async to true, and set username and password to null. Note: Unfortunately legacy content prevents treating the async argument being undefined identical from it being omitted.

and step 12

> Set the synchronous flag, if async is false, and unset the synchronous flag otherwise.

I could be wrong though, in which case I will change the below note.

Reply:
Before you get to that algorithm, the algorithm @tobie pointed out applies due to it using IDL. And that will convert the arguments to strings and booleans.

Reply:
@annevk Thanks. Updated.